### PR TITLE
k8s: add useragent

### LINKED ISSUE #7634: Set user agent for k8s clients
Set a static user agent for the k8s client that we are using. We should add cilium's commit and version to that user agent

### PR DESCRIPTION
**Pull request by pyaillet**
Signed-off-by: Pierre-Yves Aillet <pyaillet@gmail.com>

This PR adds a UserAgent in k8s requests.
The format of the useragent is `Cilium 0.13.90 a722bdb 2018-01-09T22:32:37+01:00 go version go1.9 linux/amd64` for example

Fixes: #7634

```release-note
Add `Cilium <VERSION>` as k8s useragent
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7791)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Both CI runs hit #7615 : 

* https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1923/
* https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/11748/

**Comment by ianvernon** [comment]
Going to wait for @aanm to check to ensure this is what he wants since he filed the initial issue.

**Review by aanm — APPROVED** [review]

File: pkg/k8s/client.go [hunk 1]
```diff
@@ -54,7 +55,8 @@ func createConfig(endpoint, kubeCfgPath string) (*rest.Config, error) {
 		return clientcmd.BuildConfigFromFlags("", kubeCfgPath)
 	}
 
-	config := &rest.Config{Host: endpoint}
+	userAgent := fmt.Sprintf("Cilium %s", version.Version)
```
Comment (aanm) [inline pkg/k8s/client.go]: The user agent also needs to be added to the configuration returned on line 51 and 55

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
@pyaillet thanks for the change, please take a look at my change requests

**Comment by nebril** [comment]
test-me-please

**Review by nebril — APPROVED** [review]

**Review by vadorovsky — APPROVED** [review]
Looks good, thanks for your contribution!

**Review by aanm — APPROVED** [review]

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by nirmoy** [comment]
test-me-please

**Comment by ianvernon** [comment]
Build failure: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/11882/

```
20:01:09  K8sChaosTest Restart with long lived connections 
20:01:09    TCP connection is not dropped when cilium restarts
20:01:09    /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:410
20:01:11  STEP: Deleting all cilium pods
20:01:11  STEP: Waiting cilium pods to terminate
20:02:04  STEP: Waiting for cilium pods to be ready
20:05:56  === Test Finished at 2019-04-25T20:05:56Z====
20:26:04  ===================== TEST FAILED =====================
20:28:19  cmd: kubectl get pods -o wide --all-namespaces
20:28:19  Exitcode: 0 
20:28:19  Stdout:
20:28:19   	 NAMESPACE     NAME                                    READY     STATUS             RESTARTS   AGE       IP              NODE
20:28:19  	 default       netperf-client                          1/1       Running            0          27m       10.10.0.173     k8s1
20:28:19  	 kube-system   cilium-6pqfx                            0/1       NodeLost           0          26m       192.168.36.12   k8s2
20:28:19  	 kube-system   cilium-etcd-749wglg64p                  1/1       Unknown            0          43m       10.10.1.74      k8s2
20:28:19  	 kube-system   cilium-etcd-bslw9kzqz2                  1/1       Unknown            0          43m       10.10.1.181     k8s2
20:28:19  	 kube-system   cilium-etcd-j4xsjdvpcm                  1/1       Running            0          43m       10.10.0.7       k8s1
20:28:19  	 kube-system   cilium-etcd-operator-77d4ddf8c6-n47fb   1/1       Unknown            0          34m       192.168.36.12   k8s2
20:28:19  	 kube-system   cilium-etcd-operator-77d4ddf8c6-rrnfp   0/1       CrashLoopBackOff   6          21m       192.168.36.11   k8s1
20:28:19  	 kube-system   cilium-kbrlq                            1/1       Running            6          26m       192.168.36.11   k8s1
20:28:19  	 kube-system   cilium-operator-5fb956fc65-2wjx2        0/1       CrashLoopBackOff   9          21m       10.10.0.230     k8s1
20:28:19  	 kube-system   etcd-k8s1                               1/1       Running            0          55m       192.168.36.11   k8s1
20:28:19  	 kube-system   etcd-operator-65476dd78f-whfg5          1/1       Unknown            0          34m       10.10.1.127     k8s2
20:28:19  	 kube-system   kube-apiserver-k8s1                     1/1       Running            0          54m       192.168.36.11   k8s1
20:28:19  	 kube-system   kube-controller-manager-k8s1            1/1       Running            0          54m       192.168.36.11   k8s1
20:28:19  	 kube-system   kube-dns-f4d788bb7-5xdb4                3/3       Running            0          21m       10.10.0.183     k8s1
20:28:19  	 kube-system   kube-dns-f4d788bb7-fzx4z                3/3       Unknown            0          34m       10.10.1.173     k8s2
20:28:19  	 kube-system   kube-proxy-pg7s6                        1/1       NodeLost           0          45m       192.168.36.12   k8s2
20:28:19  	 kube-system   kube-proxy-x8kj9                        1/1       Running            0          55m       192.168.36.11   k8s1
20:28:19  	 kube-system   kube-scheduler-k8s1                     1/1       Running            0          55m       192.168.36.11   k8s1
20:28:19  	 
20:28:19  Stderr:
20:28:19   	 
20:28:19  
20:30:20  cmd: kubectl exec -n kube-system cilium-6pqfx -- cilium service list
20:30:20  Exitcode: 1 
20:30:20  Stdout:
20:30:20   	 
20:30:20  Stderr:
20:30:20   	 Error from server: error dialing backend: dial tcp 192.168.36.12:10250: getsockopt: connection timed out
20:30:20  	 
20:30:20  
20:32:36  cmd: kubectl exec -n kube-system cilium-6pqfx -- cilium endpoint list
20:32:36  Exitcode: 1 
20:32:36  Stdout:
20:32:36   	 
20:32:36  Stderr:
20:32:36   	 Error from server: error dialing backend: dial tcp 192.168.36.12:10250: getsockopt: connection timed out
20:32:36  	 
20:32:36  
20:32:36  cmd: kubectl exec -n kube-system cilium-kbrlq -- cilium service list
20:32:36  Exitcode: 1 
20:32:36  Stdout:
20:32:36   	 
20:32:36  Stderr:
20:32:36   	 error: unable to upgrade connection: container not found ("cilium-agent")
20:32:36  	 
20:32:36  
20:32:36  cmd: kubectl exec -n kube-system cilium-kbrlq -- cilium endpoint list
20:32:36  Exitcode: 1 
20:32:36  Stdout:
20:32:36   	 
20:32:36  Stderr:
20:32:36   	 error: unable to upgrade connection: container not found ("cilium-agent")
20:32:36  	 
20:32:36  
20:45:54  Sending interrupt signal to process
20:46:03  cleaning up VMs started for k8s tests
20:46:03  
```

Haven't ever seen this! Looks like the VM got into a bad state in the testing infra.

**Comment by ianvernon** [comment]
test-me-please

**Comment by aanm** [comment]
test-upstream-k8s

**Comment by aanm** [comment]
test-k8s-upstream

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
Great work @pyaillet

**Pull request merged**